### PR TITLE
[SA-18200] Various improvements to Helm chart

### PR DESCRIPTION
**Pull request by mandopaloooza**
* Install sdp-k8s-client resources to `sdp-system` namespace
* Improve uninstallation for a cleaner removal of resources
* Improve post-install instructions in NOTES.txt
* Bump chart to 0.1.1

```
$ helm install sdp-k8s-client k8s/chart/
NAME: sdp-k8s-client
LAST DEPLOYED: Tue Jan 18 14:59:40 2022
NAMESPACE: default
STATUS: deployed
REVISION: 1
TEST SUITE: None
NOTES:
----------------------------------------------------------------------------------------------------------
|     ###    ########  ########   ######      ###    ######## ########     ######  ########  ########    |
|    ## ##   ##     ## ##     ## ##    ##    ## ##      ##    ##          ##    ## ##     ## ##     ##   |
|   ##   ##  ##     ## ##     ## ##         ##   ##     ##    ##          ##       ##     ## ##     ##   |
|  ##     ## ########  ########  ##   #### ##     ##    ##    ######       ######  ##     ## ########    |
|  ######### ##        ##        ##    ##  #########    ##    ##                ## ##     ## ##          |
|  ##     ## ##        ##        ##    ##  ##     ##    ##    ##          ##    ## ##     ## ##          |
|  ##     ## ##        ##         ######   ##     ##    ##    ########     ######  ########  ##          |
----------------------------------------------------------------------------------------------------------

Thank you for installing sdp-k8s-client
Your release is named sdp-k8s-client

### Getting Started

1. Create and label the namespace for SDP client injection
$ kubectl create namespace sdp-demo
$ kubectl label namespace sdp-demo --overwrite sdp-injection=enabled


2. Create a secret containing username, password, and profile URL for SDP Injector authentication
$ kubectl create secret generic sdp-injector-client-secrets \
    --namespace sdp-demo \
    --from-literal=client-username=<USERNAME> \
    --from-literal=client-password=<PASSWORD> \
    --from-literal=client-controller-url=<PROFILE_URL>


3. Create a configmap containing SDP Injector configuration
$ kubectl create configmap sdp-injector-client-config \
    --namespace sdp-demo \
    --from-literal=client-log-level=Debug

The following configurations are currently supported:
* client-log-level - The log level of the client
* client-device-id - Specific device ID in UUID v4 format to use for the client. If unspecified, a random UUID is generated


4. Create a busybox pod and ping an IP address protected by SDP
$ kubectl run --namespace sdp-demo -i --tty busybox --image=busybox -- sh
$ / # ping <IP_ADDRESS>
```

### CODE REVIEW

**Review by kajes — APPROVED** [review]
One suggestion. Otherwise LGTM.

File: k8s/chart/templates/_helpers.tpl [hunk 1]
```diff
@@ -80,4 +80,11 @@ Default app version
 */}}
 {{- define "sdp-k8s-client.defaultTag" -}}
   {{- default .Chart.AppVersion .Values.global.image.tag }}
+{{- end -}}
+
+{{/*
+Namespace
+*/}}
+{{- define "sdp-k8s-client.namespace" -}}
+  {{ print "sdp-system" }}
```
[the remark this answers is not in the document]
Comment (mandopaloooza) [reply k8s/chart/templates/_helpers.tpl:89]: Agree! Do you think it should override from .Release.Namespace? 

I need to check if anything passed in via `--namespace` overrides the `sdp-system`

[the remark this answers is not in the document]
Comment (mandopaloooza) [reply k8s/chart/templates/_helpers.tpl:89]: Ok. Just checked. `--namespace` does not override the namespace declared in the templates.

File: k8s/chart/templates/_helpers.tpl [hunk 2]
```diff
@@ -80,4 +80,15 @@ Default app version
...
+{{/*
+Namespace
+*/}}
+{{- define "sdp-k8s-client.namespace" -}}
+{{- if eq .Release.Namespace "default" }}
+{{- print "sdp-system" }}
+{{- else}}
+{{- .Release.Namespace }}
+{{- end }}
 {{- end -}}
```
Comment (lepiolet) [inline k8s/chart/templates/_helpers.tpl:94]: newline

File: k8s/chart/templates/rbac.yaml [hunk 3]
```diff
@@ -3,34 +3,34 @@ apiVersion: rbac.authorization.k8s.io/v1
...
     "helm.sh/hook-weight": "1"
 rules:
   - apiGroups: [""]
-    resources: ["secrets"]
+    resources: ["secrets", "configmaps" ]
     verbs: ["get", "create", "delete"]
   - apiGroups: ["admissionregistration.k8s.io"]
     resources: ["mutatingwebhookconfigurations/v1"]
-    verbs: ["get", "create"]
+    verbs: ["get", "create", "delete"]
```
Comment (lepiolet) [inline k8s/chart/templates/rbac.yaml:18]: ahh we were missing the delete :D

**Review by lepiolet — APPROVED** [review]

File: k8s/chart/templates/_helpers.tpl (hunk 1 above)
[the remark this answers is not in the document]
Comment (lepiolet) [reply k8s/chart/templates/_helpers.tpl]: so the way to override it is uing `--namespace` right, not using `--set namespace=blah`?

**Pull request merged**